pages/resources/quizzes/firstProgrammingLanguage/index.tsx

```diff
@@ -35,14 +42,59 @@ const initializeQuizViewState = {
 
 const FirstProgrammingLanguageQuiz = () => {
   const [stage, setStage] = useState(Stage.START);
+  const [quizResponseId, setQuizResponseId] = useState<number>();
   const [quizViewState, setQuizViewState] = useState<QuizViewState>(
     initializeQuizViewState
   );
   const [triggerAnimation, setTriggerAnimation] = useState(true);
+  const [userInput, setUserInput] = useState<UserInput>({
+    name: "",
+    email: "",
+  });
+
+  //Insert mutation - Store Name & Email. Returns id.
+  const [createQuizResponse] = useMutation(
+    INSERT_CODING_LANGUAGE_QUIZ_RESPONSE,
+    {
+      onCompleted: (data) => {
+        if (!quizResponseId) {
+          setQuizResponseId(
+            parseInt(data.insert_coding_language_quiz.returning[0].id)
+          );
+        }
+      },
+    }
+  );
+
+  //Update Mutation - Store the rest of the columns via id.
+  const [finalQuizResponse] = useMutation(UPDATE_CODING_LANGUAGE_QUIZ_RESPONSE);
```

Comment:
I would rename this to updateQuizResponse. 

finalQuizResponse is a noun, which makes me think that the `useMutation` hook is returning an object, but it's not it's returning a function that will do something, so we should use a verb instead.